source/lib/cdk-textract-stack.ts

```diff
@@ -122,6 +127,276 @@ export class CdkTextractStack extends cdk.Stack {
+          resources: [
+            covidDataBucket.bucketArn,
+            `${covidDataBucket.bucketArn}/*`
+          ]
+        })
+      );
+
+      kendraRoleArn = kendraRole.roleArn;
+
+      const onEventCreateKendraIndexLambda = new lambda.Function(this, this.resourceName('OnEventKendraIndexCreator'), {
+        code: lambda.Code.fromAsset('lambda/kendraIndexCreator'),
+        description: 'onEvent handler for creating Kendra index',
+        runtime: lambda.Runtime.PYTHON_3_7,
+        handler: 'lambda_function.lambda_handler',
+        timeout: Duration.minutes(15),
+        environment: {
+          KENDRA_ROLE_ARN: kendraRole.roleArn,
+          KMS_KEY_ID: kendraKMSKey.keyId,
+          KENDRA_INDEX_CLIENT_TOKEN: this.uuid.toLowerCase()
+        }
+      });
+
+      kendraKMSKey.grantEncryptDecrypt(onEventCreateKendraIndexLambda);
+      onEventCreateKendraIndexLambda.addLayers(boto3Layer);
+
+      onEventCreateKendraIndexLambda.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["kendra:CreateIndex","kendra:DeleteIndex","kendra:DescribeIndex","kendra:TagResource","kms:ListKeys",
+          "kms:ListAliases","kms:DescribeKey","kms:CreateGrant"],
+          resources: ["*"]
+        })
+      );
+
+      onEventCreateKendraIndexLambda.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["iam:PassRole"],
+          resources: [kendraRole.roleArn]
+        })
+      );
+
+      const isCompleteKendraIndexLambda = new lambda.Function(this, this.resourceName('isCompleteKendraIndexPoller'),{
+        code: lambda.Code.fromAsset('lambda/kendraIndexPoller'),
+        description: 'isComplete handler to check for Kendra Index creation',
+        runtime: lambda.Runtime.PYTHON_3_7,
+        handler: 'lambda_function.lambda_handler',
+        timeout: Duration.minutes(15),
+      });
+
+      isCompleteKendraIndexLambda.addLayers(boto3Layer);
+      //kendraKMSKey.grantEncryptDecrypt(isCompleteKendraIndexLambda);
+
+      isCompleteKendraIndexLambda.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["kendra:DescribeIndex"],
+          resources: ["*"]
+        })
+      );
+
+      isCompleteKendraIndexLambda.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: ["iam:PassRole"],
+          resources: [kendraRole.roleArn]
+        })
+      );
+
+      const kendraIndexProvider = new cr.Provider(this, this.resourceName('KendraIndexProvider'), {
+        onEventHandler: onEventCreateKendraIndexLambda,
+        isCompleteHandler: isCompleteKendraIndexLambda,
+        totalTimeout: Duration.hours(1),
+        queryInterval: Duration.minutes(1),
+      });
+
+      createKendraIndexCustomResource = new CustomResource(this, this.resourceName('CreateKendraIndexCustomResource'), { 
```

Comment:
Can we call this KendraIndexCustomResource?
Doesnt This take care of more than just create?